esmvaltool/cmorizers/obs/utilities.py

```diff
@@ -77,13 +77,21 @@ def fix_coords(cube):
             if cube_coord.ndim == 1:
                 if cube_coord.points[0] < 0. and \
                         cube_coord.points[-1] < 181.:
-                    cube_coord.points = \
-                        cube_coord.points + 180.
+                    lon_coord = cube.coord('longitude').copy()
+
+                    lons_below_0 = lon_coord.points[lon_coord.points < 0.] + \
+                        360.
+                    lons_above_0 = lon_coord.points[lon_coord.points >= 0.]
+                    lons = np.hstack((lons_above_0, lons_below_0))
+
```

Comment:
why is this needed? it also appears to be wrong: say we lave lons [-180 ... 180] this will convert them to [180 ... 360, 0 ...180]

Reply:
E-Obs only covers Europe, longitudes range e.g. from -24.95 to 45.45. Simply splitting them in two and adding 180° to the lower half will produce wrong values.

Reply:
OK but changing `utilties.py` will mean all the other cmorizers that import the functionality will not work properly. Maybe you will want to write a specific function to deal with this in the cmorizer script please?

Reply:
What about adding an if statement checking if longitudes <> 0 have the same length? The current floor division in _roll_cube_data(cube, nlon // 2, -1) could also hold its perils.

Reply:
Let's just go for the specific function in this case :+1:

Reply:
Should I include it in the cmorizer (since it's only used here) or the utilities?

Reply:
If it's only used in a single cmorizer, that would be the best place to put it. If another cmorizer needs it as well in the future, that would be a good time to move it to the utilities file.